client/ayon_core/tools/publisher/publish_report_viewer/widgets.py

```diff
@@ -294,6 +294,70 @@ def _set_logs(self, logs):
         self._output_widget.setPlainText(text)
 
 
+class ClassInfoWidget(QtWidgets.QWidget):
+    def __init__(self, parent):
+        super(ClassInfoWidget, self).__init__(parent)
+
+        vertical_layout = QtWidgets.QVBoxLayout()
+        vertical_layout.setContentsMargins(8, 8, 8, 8)
+        self.setLayout(vertical_layout)
+
+        form_layout = QtWidgets.QFormLayout()
+        vertical_layout.addLayout(form_layout)
+
+        self._textedit_doc = QtWidgets.QTextEdit()
+        self._textedit_doc.setReadOnly(True)
+        self._textedit_doc.setMaximumHeight(300)
+        form_layout.addRow("Documentation", self._textedit_doc)
+
+        self._lineedit_file_path = QtWidgets.QLineEdit()
+        self._lineedit_file_path.setReadOnly(True)
+        form_layout.addRow("File Path", self._lineedit_file_path)
+
+        self._lineedit_plugin_type = QtWidgets.QLineEdit()
+        self._lineedit_plugin_type.setReadOnly(True)
+        form_layout.addRow("Plugin type", self._lineedit_plugin_type)
+
+        # self._lineedit_instances_labels = QtWidgets.QLineEdit()
+        # self._lineedit_instances_labels.setReadOnly(True)
+        # form_layout.addRow("Runs on instances", self._lineedit_instances_labels)
+
+        self._lineedit_families = QtWidgets.QLineEdit()
+        self._lineedit_families.setReadOnly(True)
+        form_layout.addRow("Families", self._lineedit_families)
+
+        vertical_layout.addStretch(100)
```

Comment:
Few notes about widgets in host tools and tools:
- Use parenting if possible, some hosts have different behavior of style propagation when parent is not passed on `__init__`. It also applies to adding layout to layout, in that case the styles are not sometimes propagated to children widgets.
- If possible, store all widgets at the end of `__init__` so you can easily see what the widget actually has. When someone else will look in future he should look at one place, going through each line of `__init__` is hell. Also define layout at one place so you can see what is set and what is added to the layout.
```suggestion
        info_widget = QtWidgets.QWidget(self)

        plugin_doc_widget = QtWidgets.QTextEdit(info_widget)
        plugin_doc_widget.setReadOnly(True)
        plugin_doc_widget.setMaximumHeight(300)

        plugin_path_widget = QtWidgets.QLineEdit(info_widget)
        plugin_path_widget.setReadOnly(True)

        plugin_type_widget = QtWidgets.QLineEdit(info_widget)
        plugin_type_widget.setReadOnly(True)

        plugin_families_widget = QtWidgets.QLineEdit(info_widget)
        plugin_families_widget.setReadOnly(True)

        info_layout = QtWidgets.QFormLayout(info_widget)
        info_layout.setContentsMargins(0, 0, 0, 0)
        info_layout.addRow("Documentation", plugin_doc_widget)
        info_layout.addRow("File Path", plugin_path_widget)
        info_layout.addRow("Plugin type", plugin_type_widget)
        info_layout.addRow("Families", plugin_families_widget)

        main_layout = QtWidgets.QVBoxLayout(self)
        main_layout.setContentsMargins(8, 8, 8, 8)
        main_layout.addWidget(info_widget, 0)
        main_layout.addStretch(1)

        self._plugin_doc_widget = plugin_doc_widget
        self._plugin_path_widget = plugin_path_widget
        self._plugin_type_widget = plugin_type_widget
        self._plugin_families_widget = plugin_families_widget
```